src/boss/rbo3/rbo3.c

```diff
@@ -1,13 +1,499 @@
+                    g_Tilemap.x = 256;
+                    g_Tilemap.left++;
+                    self->step++;
+                }
+                break;
+            }
+        }
+        break;
+    }
+}
+
+extern Entity D_8007A958;
+extern Entity D_8007C0D8;
+extern u16 D_us_80180480;
+extern u8 D_us_801805F0;
+extern u8 D_us_801805FC;
+extern u8 D_us_8018060C;
+extern u8 D_us_80180618;
+extern u8 D_us_80180624;
+extern u8 D_us_80180630;
+extern u8 D_us_8018063C;
+extern s8 D_us_80180648[];
+extern u8 D_us_80180670[];
+extern s32 D_us_80180728;
+
+void EntityMedusa(Entity* self) {
+    Entity* entity;
+    s16 rotZ;
+    s32 x;
+    s32 i;
+    s32 velocityX;
+    s8* rect;
+    u8* indexes;
+
+    if (self->flags & 0x100) {
+        if (self->step != 7) {
+            SetStep(7);
+        }
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(&D_us_80180480);
+        self->animCurFrame = 1;
+        self->hitboxState = 0;
+        CreateEntityFromEntity(UNK_ENTITY_25, self, self + 1);
+        SetStep(1);
+        // fallthrough
+
+    case 1:
+        if (D_us_80180728 & 1) {
+            SetStep(2);
+        }
+        break;
+
+    case 2:
+        // n.b.! AnimateEntity is not declared
+        if (!AnimateEntity(&D_us_801805F0, self)) {
+            self->hitboxState = 3;
+            SetStep(3);
+        }
+        break;
+
+    case 3:
+        AnimateEntity(&D_us_801805FC, self);
+        if (self->step_s == 0) {
+            self->ext.GS_Props.timer = 64;
+            self->step_s++;
+        }
+        if (GetDistanceToPlayerX() < 104) {
+            self->ext.GS_Props.attackMode = 1;
+        }
+        if (GetDistanceToPlayerX() > 128) {
+            self->ext.GS_Props.attackMode = 0;
+        }
+        if (GetDistanceToPlayerX() > 32) {
+            self->facingLeft = GetSideToPlayer() & 1;
+        }
+        MoveEntity();
+
+        if (self->facingLeft == self->ext.GS_Props.attackMode) {
+            self->velocityX = FIX(1.0 / 2.0);
+        } else {
+            self->velocityX = -FIX(1.0 / 2.0);
+        }
+        if (self->hitFlags & 3) {
+            SetStep(6);
+        }
+        x = PLAYER.posX.i.hi - self->posX.i.hi;
+        if (g_Player.status & PLAYER_STATUS_UNK2000 &&
+            (x * PLAYER.velocityX) < 0) {
+            if (abs(x) < 80) {
+                SetStep(5);
+            }
+        }
+
+        if (!--self->ext.GS_Props.timer) {
+            GetSideToPlayer();
+            if (GetDistanceToPlayerX() <= 64) {
+                SetStep(5);
+            } else {
+                SetStep(4);
+            }
+        }
+        break;
+    case 5:
+        if (self->step_s == 0) {
+            if (Random() & 1) {
+                PlaySfxPositional(SFX_RBO3_UNK_7FE);
+            } else {
+                PlaySfxPositional(SFX_RBO3_UNK_7FF);
+            }
+            PlaySfxPositional(SFX_RBO3_UNK_7C8);
+            self->step_s++;
+        }
+        self->ext.stub[9] = 1;
```

Comment:
I think mixing the extensions is going to be problematic since they're not necessarily guaranteed to align

Reply:
Eek. Missed that one. Fixed.